server/src/main/java/org/opensearch/action/admin/cluster/remotestore/restore/RestoreRemoteStoreRequest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.restore;
+
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Restore remote store request
+ *
+ * @opensearch.internal
+ */
+public class RestoreRemoteStoreRequest extends ClusterManagerNodeRequest<RestoreRemoteStoreRequest> implements ToXContentObject {
+
+    private String[] indices = Strings.EMPTY_ARRAY;
+    private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpen();
+    private boolean waitForCompletion;
+
+    public RestoreRemoteStoreRequest() {}
```

Comment:
Shouldn't indices be in constructor args(mandatory)

Reply:
I meant just from a stand alone class perspective

Reply:
Got it. It is a bit tricky as it depends on how `RestRestoreRemoteStoreAction` initializes the `RestoreRemoteStoreRequest`.

https://github.com/opensearch-project/OpenSearch/pull/3576/files#diff-03a57182daff0959e24e2fccf7fe28e7dedcd711ef76758e24df834dc0e884b7R41

As indices is a part of post body, fetching it from RestRequest requires parsing the body content. This code of parsing is added as part of `source` method of the same class.

I just followed the conventions used by other Request classes but open to suggestions.